eclair-core/src/main/scala/fr/acinq/eclair/payment/Relayer.scala

```diff
@@ -74,7 +74,14 @@ class Relayer(nodeParams: NodeParams, register: ActorRef, paymentHandler: ActorR
   def main(channelUpdates: Map[ShortChannelId, OutgoingChannel], node2channels: mutable.HashMap[PublicKey, mutable.Set[ShortChannelId]] with mutable.MultiMap[PublicKey, ShortChannelId]): Receive = {
 
     case GetUsableBalances =>
-      sender ! channelUpdates.values.map(o => UsableBalances(o.commitments.availableBalanceForSendMsat, o.commitments.availableBalanceForReceiveMsat, o.commitments.announceChannel))
+      sender ! channelUpdates.values
+          .filter(o => Announcements.isEnabled(o.channelUpdate.channelFlags))
+          .map(o => UsableBalances(
+            canSendMsat = if (o.commitments.availableBalanceForSendMsat < 0) 0 else o.commitments.availableBalanceForSendMsat,
```

Comment:
`availableBalanceForSendMsat ` and `availableBalanceForReceiveMsat ` should not return negative values imo. Also this is better implemented with a `max`.